src/shims/posix/foreign_items.rs

```diff
@@ -67,7 +67,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 let buf = this.read_scalar(buf)?.not_undef()?;
                 let count = this.read_scalar(count)?.to_machine_usize(this)?;
                 let result = if fd == 0 {
-                    throw_unsup_format!("reading from stdin is not implemented")
+                    use std::io::{self, Read};
+
+                    this.check_no_isolation("read")?;
+
+                    let mut buffer = vec![0; count as usize];
```

Comment:
`as` casts are always suspicious. Why can this not be bigger than a host usize?
This is what the `read` for files does:
```rust
// We cap the number of read bytes to the largest value that we are able to fit in both the
// host's and target's `isize`. This saves us from having to handle overflows later.
let count = count.min(this.machine_isize_max() as u64).min(isize::MAX as u64);
```